narwhal/primary/src/proposer.rs

```diff
@@ -640,7 +657,16 @@ impl Proposer {
                         Ordering::Equal => {
                             // The core gives us the parents the first time they are enough to form a quorum.
                             // Then it keeps giving us all the extra parents.
-                            self.last_parents.extend(parents)
+                            self.last_parents.extend(parents);
+
+                            // As the schedule can change after an odd round proposal - when the new schedule algorithm is
+                            // enabled - make sure that the timer is reset correctly for the round leader. No harm doing
+                            // this here as well.
+                            if self.min_delay() == Duration::ZERO {
+                                min_delay_timer
+                                .as_mut()
+                                .reset(Instant::now());
+                            }
```

Comment:
@mwtian @arun-koshy same as above. This won't affect the current version, but will guarantee that in the even of schedule change where the next leader changes, the right leader will still propose early enough